### PR TITLE
Expense app: Update expense up

### PR DESCRIPTION
**Pull request by ShellyGolden**
- [x] Edit models.py file
- [x] Add migrations/0001_initial.py file
- [x] Add test file

Signed-off-by: Shelly Golden <shelly.golden@gmail.com>

### CODE REVIEW

File: expenses/tests_expenses.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+@pytest.fixture
+def expense_test0():
+    return Expenses(amount=AMOUNT,
+                    date=DATE,
+                    Category=CATEGORY
+                    )
+
+
+@pytest.mark.django_db
+class TestExpenseדModel:
```
Comment (shlior7) [inline expenses/tests_expenses.py:19]: typo _ד_

**Comment by ShellyGolden** [comment]
I have a little problem, I'll take care of tomorrow

File: expenses/tests_expenses.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+def test_save_expense(self, expense_test0):
+    expense_test0.save()
+    assert expense_test0 in Expenses.objects.all()
+
+
+def test_delete_expense(self, expense_test0):
+    expense_test0.save()
+    expense_test0.delete()
+    assert expense_test0 not in Expenses.objects.all()
```
Comment (shlior7) [inline expenses/tests_expenses.py:34]: are these suppose to be in the class TestExpenseModel?(self in params)
if so they are missing an indentation

Comment (shlior7) [reply expenses/tests_expenses.py:34]: also I like the idea of testing the save and delete separately ill implement that in the House app as well, good one 🙌

**Comment by ShellyGolden** [comment]
This PR blocked by https://github.com/redhat-beyond/Xpense/pull/28

**Comment by svatares** [comment]
@ShellyGolden #28 was merged, please rebase

File: expenses/tests_expenses.py (hunk 2 above)
Comment (ShellyGolden) [reply expenses/tests_expenses.py]: fixed

> are these suppose to be in the class TestExpenseModel?(self in params) if so they are missing an indentation

**Comment by ShellyGolden** [comment]
Hey team, 
please take a look :)

**Comment by ShellyGolden** [comment]
> @ShellyGolden #28 was merged, please rebase

done

File: expenses/models.py [hunk 3]
```diff
@@ -1,7 +1,12 @@
 from django.db import models
+from django.utils import timezone
 
 
 class Expenses(models.Model):
+    # house_id = models.ForeignKey('user_id', on_delete=models.CASCADE)
```
Comment (adiabramovitch) [inline expenses/models.py:6]: What about this comment?

File: expenses/tests_expenses.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+DATE = timezone.now()
+CATEGORY = 'CLOTHING'
+
+
+@pytest.fixture
+def expense_test0():
+    return Expenses(amount=AMOUNT,
+                    date=DATE,
+                    Category=CATEGORY
+                    )
```
Comment (adiabramovitch) [inline expenses/tests_expenses.py:15]: Maybe add tests regarding the types and values of an Expense? 
What happens if you create an expense with a non-valid category? 
Does the date of an expense count valid if it is the future? 
Can the amount be negative? 
All those and much more should be tested and validated

File: expenses/models.py (hunk 3 above)
Comment (ShellyGolden) [reply expenses/models.py:6]: > What about this comment?

I'll be glad if you can help me with the connection between expense and house (expense needs the house_id of it's house)
thanks! :) 
@adiabramovitch

File: expenses/tests_expenses.py (hunk 4 above)
Comment (ShellyGolden) [reply expenses/tests_expenses.py:15]: > Maybe add tests regarding the types and values of an Expense? What happens if you create an expense with a non-valid category? Does the date of an expense count valid if it is the future? Can the amount be negative? All those and much more should be tested and validated

There is a default category - other.
I'm taking care about of the others
@adiabramovitch

File: expenses/models.py (hunk 3 above)
Comment (ronyyosef) [reply expenses/models.py:6]: Are we going to remove this comment when the House model will be merged to master?
I think it is there already: https://github.com/redhat-beyond/Xpense/blob/main/house/models.py
and you can try to remove it and test the interaction between house and expenses

**Review by ronyyosef — COMMENTED** [review]
Great work
I left a comment about the integration with the house model
Please update us if you get any errors while creating a house with expenses

File: expenses/models.py [hunk 5]
```diff
@@ -23,4 +31,23 @@ class Category(models.TextChoices):
...
-        return self.category
+        return f'Category:{self.category},Amount:{self.amount}'
+
+    def validate_amount(amount):
+        if amount is not None:
+            if amount <= 0:
+                raise ValidationError(
+                    _(f'{amount} is not a valid amount. The amount should be bigger then 0'),
+                    params={'amount': amount},
+                )
```
[the remark this answers is not in the document]
Comment (ShellyGolden) [reply expenses/models.py:42]: thanks!

File: expenses/tests_expenses.py (hunk 4 above)
Comment (ShellyGolden) [reply expenses/tests_expenses.py:15]: For now, all those things are expressed in the model. 
If we see it’s important, in the future, we will add it.

**Review by Yoavsc129 — APPROVED** [review]

File: expenses/models.py (hunk 3 above)
Comment (ShellyGolden) [reply expenses/models.py]: I fixed it

**Review by LiorKGOW — APPROVED** [review]
LGTM

**Comment by adiabramovitch** [comment]
@ShellyGolden  you have conflicts to resolve

**Comment by ShellyGolden** [comment]
> @ShellyGolden you have conflicts to resolve

fixed

**Review by shlior7 — APPROVED** [review]

**Review by LiorKGOW — APPROVED** [review]
LGTM

**Review by ronyyosef — APPROVED** [review]

**Review by Yoavsc129 — APPROVED** [review]

**Comment by ronyyosef** [comment]
Hey,
Can someone merge it please?

**Review by ronyyosef — APPROVED** [review]

**Pull request merged**